docs/network-spec/connection-manager.tex

```diff
@@ -273,7 +356,212 @@ \section{Components}
+cleanly. When it does terminate cleanly the governor restarts the mini-protocol
+handler.
+
+All the mini-protocols have the property that agency starts with the remote
+peer\footnote{Originally transaction submission protocol had agency start with the
+server side. A later protocol update reversed the initial agency so that they
+are now all consistent.}. This allows all of the server side protocols to be
+started in the mux `on-demand' mode. In the on-demand mode the protocol handler
+thread is not started until the first message arrives from the client.
+\todo[inline]{this lack of active processing locally is used as a witness of remote
+      peer idleness - where should we be explicit about that?}
+
+The inbound governor gets informed of new connections that it should monitor
+either via by the server or by the connection manager. The server informs the
+governor about fresh inbound connections. The connection manager informs the
+governor about connections that it started due to a request for an outbound
+connection -- at least for those connections that are to be available to use in
+duplex mode.
+
+\begin{designalternative}
+As illustrated in \cref{tik:components}, both the connection manager and server
+components communicate with the inbound governor directly. They do this to
+inform the inbound governor about new connections so that it can start to run
+and monitor the server side protocols. The server notifies about new connections
+established inbound, while the connection manager notifies about new connections
+established outbound (at least the duplex ones).
+A slight simplification would be to have only one of these routes of
+notification. The more obvious choice would be for the connection manager to
+inform the inbound governor about all new connections. Then the server would
+only interact with the connection manager and not with the inbound governor.
+
+\begin{center}
+\begin{tikzpicture}
+  \footnotesize
+  \def\xa{-2.0}
+  \def\xb{2.0}
+
+  \node[rounded corners, rectangle, draw, anchor=east, text width=3.5cm] (p2p_governor) at (\xa, -2.5)
+   {
+     \hfill\textbf{Peer-to-Peer Governor}\hfill
+     \vspace{5pt}
+    };
+
+  \node[rounded corners, rectangle, draw, anchor=west, text width=3cm] (server) at (\xb, -1.50)
+    {
+      \hfill{\textbf{Server}}\hfill
+      \vspace{5pt}
+    };
+
+  \node[rounded corners, rectangle, draw, anchor=west, text width=3cm] (inbound_governor) at (\xb, -3.5)
+    {
+      \hfill{\textbf{Inbound Governor}}\hfill
+      \vspace{5pt}
+    };
+
+  \node[rounded corners, rectangle, draw, minimum height=1cm, text width=2cm] (connection_manager) at (0, -2.5)
+    {
+      \hfil\textbf{Connection}\hfil\\
+      \hfil\textbf{Manager}\hfil
+    };
+
+  \draw[->] (p2p_governor)           -- (connection_manager);
+  \draw[->] (server.west)            -- (connection_manager.5);
+  \draw[<->] (connection_manager.355) -- (inbound_governor.west);
+\end{tikzpicture}
+\end{center}
+\end{designalternative}
```

Comment:
Server does not interact with the inbound governor. Whenever the `Server` accepts a connection it calls `includeInboundConnection` which will notify the connection manager. Connection manager will perform handshake and inform Inbound Governor about the connection about it to start miniprotocols and will also inform the outbound governor about it if `Duplex` connection got negotiated in order for that inbound connection to be added to the `KnownPeer` set.

So in a way this design is already in place.